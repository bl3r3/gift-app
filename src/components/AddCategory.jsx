import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("one punch");

  const onIputChange = (e) => {
    setInputValue(e.target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length <= 1) return;

    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="buscar gifs"
        value={inputValue}
        onChange={onIputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
