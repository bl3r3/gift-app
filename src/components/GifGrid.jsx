import { GifItem } from "./GifItem";
import { useFetchGif } from "../hooks/useFetchGif";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGif(category);

  return (
    <>
      {isLoading ? (
        <h2>Loading...</h2>
      ) : (
        <>
          <h3>{category}</h3>
          <div className="card-grid">
            {images.map((image) => (
              <GifItem key={image.id} {...image} />
            ))}
          </div>
        </>
      )}
    </>
  );
};
