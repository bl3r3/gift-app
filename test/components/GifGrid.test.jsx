import { render, screen } from "@testing-library/react";
import { GifGrid } from "../../src/components/GifGrid";
import { useFetchGif } from "../../src/hooks/useFetchGif";

jest.mock("../../src/hooks/useFetchGif");

describe("Test in <GifGrid /> component", () => {
  const category = "One punch";

  it("debe de mostrar el loading inicialmente", () => {
    useFetchGif.mockReturnValue({
      images: [],
      isLoading: true,
    });

    render(<GifGrid category={category} />);

    expect(screen.getByText("Loading..."));
    expect(screen.getByText(category));
  });

  it("debe de mostrar items cuando carga", () => {
    const gifs = [
      {
        id: "1",
        title: "Saitama",
        url: "https://localhost/saitama.jpg",
      },
      {
        id: "2",
        title: "Rick",
        url: "https://localhost/rick.jpg",
      },
    ];

    useFetchGif.mockReturnValue({
      images: gifs,
      isLoading: false,
    });

    render(<GifGrid category={category} />);
    expect(screen.getAllByRole("img").length).toBe(2);
  });

  it("Debe hacer match con el snapshot", () => {
    const { container } = render(<GifGrid category={category} />);
    expect(container).toMatchSnapshot();
  });
});
