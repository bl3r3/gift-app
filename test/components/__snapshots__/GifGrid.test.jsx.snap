// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGrid /> component Debe hacer match con el snapshot 1`] = `
<div>
  <h3>
    One punch
  </h3>
  <div
    class="card-grid"
  >
    <div
      class="card"
    >
      <img
        alt="Saitama"
        src="https://localhost/saitama.jpg"
      />
      <p>
        Saitama
      </p>
    </div>
    <div
      class="card"
    >
      <img
        alt="Rick"
        src="https://localhost/rick.jpg"
      />
      <p>
        Rick
      </p>
    </div>
  </div>
</div>
`;
